src/server/repositories/UrlRepository.ts

```diff
@@ -193,6 +241,84 @@ export class UrlRepository implements UrlRepositoryInterface {
       })
     })
   }
+
+  private async getRelevantUrls(
+    tableName: string,
+    urlVector: string,
+    rankingAlgorithm: string,
+    limit: number,
+    offset: number,
+    query: string,
+  ) {
+    const rawQuery = `
+      SELECT ${tableName}.*
+      FROM ${tableName}, plainto_tsquery($query) query
+      WHERE query @@ (${urlVector}) AND state = '${StorableUrlState.Active}'
+      ORDER BY (${rankingAlgorithm}) desc
+      limit $limit
+      offset $offset`
+    const urlsModel = (await sequelize.query(rawQuery, {
+      bind: {
+        limit,
+        offset,
+        query,
+      },
+      type: QueryTypes.SELECT,
+      model: Url,
+      mapToModel: true,
+    })) as Array<UrlType>
+    return urlsModel
+  }
+
+  private getRankingAlgorithm(
+    order: SearchResultsSortOrder,
+    urlVector: string,
+    tableName: string,
+  ) {
+    let rankingAlgorithm
+    switch (order) {
+      case SearchResultsSortOrder.Relevance:
+        {
+          // The 3rd argument passed into ts_rank_cd represents
+          // the normalization option that specifies whether and how
+          // a document's rank should impact its rank. It works as a bit mask.
+          // 1 divides the rank by 1 + the logarithm of the document length
+          const textRanking = `ts_rank_cd(${urlVector}, query, 1)`
+          rankingAlgorithm = `${textRanking} * log(${tableName}.clicks + 1)`
```

Comment:
let's also add a recency metric for relevance ranking, when you have the time in a future PR